conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,443 @@
+            f.close()
+
+class TarGzCompressor(TarCompressor):
+    read_mode = "r:gz"
+    write_mode = "w:gz"
+    ext = ".tgz"
+
+
+class TarBz2Compressor(TarCompressor):
+    read_mode = "r:bz2"
+    write_mode = "w:bz2"
+    ext = ".tbz2"
+
+
+class TarLZMACompressor(TarCompressor):
+    read_mode = "r:xz"
+    write_mode = "w:xz"
+    ext = ".txz"
+
+
+compressors = [{"compressor": ZipCompressor()},
+               {"compressor": TarCompressor()},
+               {"compressor": TarGzCompressor()},
+               {"compressor": TarBz2Compressor()}]
+
+if sys.version_info.major >= 3:
+    compressors.append({"compressor": TarLZMACompressor()})
+
+
+# check if we can actually support symlinks:
+# some operation systems might not support them
+# e.g. Windows 10 requires either privileges or developer mode
+# also only certain file systems have support (e.g. NTFS does, but FAT32 doesn't)
+def symlinks_supported():
+    if not hasattr(os, "symlink"):
+        return False
+    client = TestClient()
+    try:
+        client.save({"a": ""})
+        with chdir(client.current_folder):
+            create_symlink("a", "b")
+            return is_link("b")
+    except OSError:
+        return False
+
+
+def hardlinks_supported():
+    if not hasattr(os, "link"):
+        return False
+    client = TestClient()
+    try:
+        client.save({"a": ""})
+        with chdir(client.current_folder):
+            os.link("a", "b")
+            return os.stat("a").st_ino == os.stat("b").st_ino
+    except OSError:
+        return False
+
+
+@parameterized_class(compressors)
+@attr("slow")
+class CompressionTest(unittest.TestCase):
+    @classmethod
+    def get_file_list(cls, d):
+        files = []
+        dirs = []
+        links = []
+        for root, dirnames, filenames in os.walk(d):
+            for filename in filenames:
+                relpath = os.path.relpath(os.path.join(root, filename), d)
+                if is_link(os.path.join(root, filename)):
+                    links.append(relpath)
+                else:
+                    files.append(relpath)
+            for dirname in dirnames:
+                relpath = os.path.relpath(os.path.join(root, dirname), d)
+                if is_link(os.path.join(root, dirname)):
+                    links.append(relpath)
+                else:
+                    dirs.append(relpath)
+        return files, dirs, links
+
+    @classmethod
+    def is_broken_link(cls, link):
+        if is_link(link):
+            target = os.readlink(link)
+            return not os.path.exists(target)
+        return False
+
+    @classmethod
+    def get_inodes(cls, d, files):
+        inodes = dict()
+        for filename in files:
+            stat = os.stat(os.path.join(d, filename))
+            if stat.st_ino not in inodes:
+                inodes[stat.st_ino] = []
+            inodes[stat.st_ino].append(filename)
+        return inodes.values()
+
+    def compare_dirs(self, d1, d2):
```

Comment:
Have you considered ``from filecmp import dircmp``?

Reply:
I'll take a look, didn't know about this one. in general I need to verify at least:
- files, directories and links (include broken links)
- hard-links (e.g. files having the same inodes)
- permissions
- timestamps

if `dircmp` can do that all, I'll use it

Reply:
not sure I understand how to use it from the docs...
e.g. for me it returns (`report_full_closure`):
```
Common funny cases : ['0a3866c1-3abb-4360-a4ad-d06e387dcaa0.6ea', '83f844bf-f6b8-4ef6-93c4-05330df32e1a.kRN']
```
what exactly does it mean? does it mean directories are identical or not?